spec/operators/buffer-spec.ts

```diff
@@ -324,6 +324,43 @@ describe('Observable.prototype.buffer', () => {
     expect(results).to.deep.equal([[1], [2], [], 'complete']);
   });
 
+  it('should buffer when Promise resolves', (done) => {
+    const e1 = interval(3).pipe(take(5));
+    const expected = [
+      [0, 1],
+      [2, 3, 4],
+    ];
+
+    e1.pipe(buffer(new Promise<void>((resolve) => setTimeout(() => resolve(), 8)))).subscribe({
```

Comment:
Nit: (Can be fixed in a FLUP)... we don't need to wait 8 ms... we can just do `Promise.resolve()`.